pegasus_wrapper/workflow.py

```diff
@@ -195,6 +213,51 @@ def get_file(self, logical_file_name: str) -> File:
             )
         return self._lfn_to_file[logical_file_name]
 
+    def _define_transformation(
+        self,
+        name: str,
+        pfn: str,
+        *,
+        namespace: Optional[str] = None,
+        version: Optional[str] = None,
+        site: Optional[str] = None,
+        is_stageable: bool = False,
+        bypass_staging: bool = True,
+        container: Optional[Container] = None,
+        arch: Optional[Arch] = None,
+        os_type: Optional[OS] = None,
+    ) -> PegasusTransformation:
+        # Try to see if we have the target transformation already made
+        if name in self._transformation_name_to_transformations:
+            for transformation in self._transformation_name_to_transformations[name]:
+                if transformation.container == container:
+                    return transformation
+        # Otherwise make the transformation and return it
+        transform = Transformation(
+            name,
+            namespace,
+            version,
+            site if site is not None else self._default_site,
+            pfn,
+            is_stageable=is_stageable,
+            bypass_staging=bypass_staging,
+            container=container,
+            arch=arch,
+            os_type=os_type,
+        )
+
+        self._transformation_catalog.add_transformations(transform)
+
+        peg_transform = PegasusTransformation(
```

Comment:
nit: `pegasus_transform` is not that much longer than `peg_transform` and increases clarity of the variable.